### DISCUSSION TITLE
About switch from @import to @use

### DISCUSSION THREAD

**Discussion by DidoMarchet** [nuxt/nuxt#29904] [General]
Hi! 
I'm trying to switch from the @import rule to the @use rule to avoid breaking changes with Dart Sass 3.0.0.

I have this configuration:

```js

vite: {
  css: {
    preprocessorOptions: {
      scss: {
        additionalData: `
          @use "@/assets/scss/vars/_index.scss" as *;
          @use "@/assets/scss/mixins/_index.scss" as *;
        `,
      },
    },
  },
}
```

But when using @use, it doesn't inject the mixins and variables globally.

So, I need to declare this in every SCSS file that uses mixins or variables:

```scss

@use '@/assets/scss/vars/_index.scss' as *; 
@use '@/assets/scss/mixins/_index.scss' as *;
```

How do you manage this switch?

I feel like I'm missing something because I can't see the global vars and mixins.

Thanks and kind regards,
Davide

**Comment by francescophra** [comment]
Hi @DidoMarchet, did you find a solution for the issue?

**Reply by DidoMarchet** [reply]
Hi @francescophra , not exactly.
I can remove the errors by adding the global variables and mixins to the Vite configuration, but they still aren’t recognized globally. This means I need to manually import the variables and mixins into each file.
It works if I use the variables or mixins directly in the template file.

It would be great to have an update or clarification regarding this setup from Nuxt team :D

**Comment by Saeid-Za** [comment]
Hello There,
There is a [related issue](https://github.com/vitejs/vite/discussions/18388) in vite repo with a workaround, could you take a look at it to see if it works?
